src/operations/bulk_write.ts

```diff
@@ -24,18 +25,15 @@ export class BulkWriteOperation extends OperationBase<BulkWriteOptions, BulkWrit
 
     this.collection = collection;
     this.operations = operations;
+
+    // Assign BSON serialize options to OperationBase, preferring options over collection options
+    Object.assign(this, inheritBSONOptions(options, collection.s.options, true));
```

Comment:
I don't think we should use `Object.assign` on `this` it does not give any information just by looking at this how `this` is being manipulated.

Reply:
Alternative examples in https://github.com/mongodb/node-mongodb-native/pull/2589#discussion_r508812071